crates/rmf_site_editor/src/site/util.rs

```diff
@@ -21,6 +21,19 @@ use bevy::prelude::*;
 use std::collections::HashSet;
 use std::path::PathBuf;
 
+pub fn line_transform(p_start: &Vec3, p_end: &Vec3) -> Transform {
+    let dp = *p_end - *p_start;
+
+    let yaw = dp.y.atan2(dp.x);
+    let tilt = dp.z.atan2(dp.x.abs());
+    let center = (*p_start + *p_end) / 2.0;
+    Transform {
+        translation: Vec3::new(center.x, center.y, center.z),
+        rotation: Quat::from_euler(EulerRot::ZYX, yaw, -tilt, 0.),
+        ..default()
+    }
+}
+
```

Comment:
This is pretty much fully duplicated with the function below. I can't quite understand, why is it necessary to remove scaling from this function and add it to the mesh instead?
I would recommend reusing the `line_stroke_transform` instead. Another useful side effect is that if you spawn the same mesh multiple times Bevy will instance it and make rendering much more efficient as shown in [this example](https://bevy.org/examples/shaders/automatic-instancing/)